packages/zoe/src/contractFacet/zcfZygote.js

```diff
@@ -0,0 +1,353 @@
+// @ts-check
+
+import { assert, details as X, makeAssert } from '@agoric/assert';
+import { E } from '@agoric/eventual-send';
+import { Far } from '@agoric/marshal';
+import { AssetKind, AmountMath } from '@agoric/ertp';
+import { makeNotifierKit, observeNotifier } from '@agoric/notifier';
+import { makePromiseKit } from '@agoric/promise-kit';
+
+import { cleanProposal } from '../cleanProposal';
+import { evalContractBundle } from './evalContractCode';
+import { makeExitObj } from './exit';
+import { objectMap } from '../objArrayConversion';
+import { makeHandle } from '../makeHandle';
+import { makeIssuerStorage } from '../issuerStorage';
+import { makeIssuerRecord } from '../issuerRecord';
+import { createSeatManager } from './zcfSeat';
+import { makeInstanceRecordStorage } from '../instanceRecordStorage';
+import { handlePWarning, handlePKitWarning } from '../handleWarning';
+import { makeOfferHandlerStorage } from './offerHandlerStorage';
+
+import '../../exported';
+import '../internal-types';
+
+import '@agoric/swingset-vat/src/types';
+
+/** @type {MakeZCFZygote} */
+export const makeZCFZygote = (
+  powers,
+  zoeService,
+  invitationIssuer,
+  testJigSetter,
+) => {
+  /** @type {PromiseRecord<ZoeInstanceAdmin>} */
+  const zoeInstanceAdminPromiseKit = makePromiseKit();
+  const zoeInstanceAdmin = zoeInstanceAdminPromiseKit.promise;
+
```

Comment:
This code is new. In contractFacet.js, we had access to the zoeInstanceAdmin presence from the start. In this version, the zoeInstanceAdmin is not available until the last step, `zcfZygote.startContract`, so we will use a promise for a presence instead.